gpjax/decision_making/decision_maker.py

```diff
@@ -0,0 +1,254 @@
+        if self.posterior_handlers.keys() != self.datasets.keys():
+            raise ValueError(
+                "Posterior handlers and datasets must have the same keys. "
+                f"Got posterior handlers keys {self.posterior_handlers.keys()} and "
+                f"datasets keys {self.datasets.keys()}."
+            )
+
+        # Initialize posteriors
+        self.posteriors: Dict[str, AbstractPosterior] = {}
+        for tag, posterior_handler in self.posterior_handlers.items():
+            self.posteriors[tag] = posterior_handler.get_posterior(
+                self.datasets[tag], optimize=True, key=self.key
+            )
+
+    def ask(self, key: KeyArray) -> Float[Array, "1 D"]:
+        """
+        Get updated acquisition function and return the point which maximises it. This
+        method also stores the acquisition function in
+        `self.current_acquisition_function` so that it can be accessed after the ask
+        function has been called. This is useful for non-deterministic acquisition
+        functions, which will differ between calls to `ask` due to the splitting of
+        `self.key`.
+
+        Args:
+            key (KeyArray): JAX PRNG key for controlling random state.
+
+        Returns:
+            Float[Array, "1 D"]: Point to be queried next.
+        """
+        self.current_acquisition_function = (
+            self.acquisition_function_builder.build_acquisition_function(
+                self.posteriors, self.datasets, key
+            )
+        )
+
+        key, _ = jr.split(key)
+        return self.acquisition_maximizer.maximize(
+            self.current_acquisition_function, self.search_space, key
+        )
+
+    def tell(self, observation_datasets: Mapping[str, Dataset], key: KeyArray):
+        """
+        Add newly observed data to datasets and update the corresponding posteriors.
+
+        Args:
+            observation_datasets (Mapping[str, Dataset]): Dictionary of datasets
+            containing new observations. Tags are used to distinguish datasets, and
+            correspond to tags in `posterior_handlers` and `self.datasets`.
+            key (KeyArray): JAX PRNG key for controlling random state.
+        """
+        if observation_datasets.keys() != self.datasets.keys():
+            raise ValueError(
+                "Observation datasets and existing datasets must have the same keys. "
+                f"Got observation datasets keys {observation_datasets.keys()} and "
+                f"existing datasets keys {self.datasets.keys()}."
+            )
+
+        for tag, observation_dataset in observation_datasets.items():
+            self.datasets[tag] += observation_dataset
+
+        for tag, posterior_handler in self.posterior_handlers.items():
+            key, _ = jr.split(key)
+            self.posteriors[tag] = posterior_handler.update_posterior(
+                self.datasets[tag], self.posteriors[tag], optimize=True, key=key
+            )
+
+    def run(
```

Comment:
could this be a method of the more abstract class as  well? It just requires ask and tell right?

Reply:
Yeah, will just add to abstract class. I'll also add a concrete implementation of `tell` to the abstract class as this should be pretty standard across different decision-making methods.